packages/smart-contracts/src/contracts/lib/SafeERC20.sol

```diff
@@ -90,4 +83,23 @@ library SafeERC20 {
     /* solium-enable security/no-inline-assembly */
     return result;
   }
+
+  
+  bytes4 private constant SELECTOR = bytes4(keccak256(bytes('transfer(address,uint256)')));
+
+  function safeTransfer(IERC20 _token, address _to, uint256 _amount) internal {
+      address tokenAddress = address(_token);
+
+      // check if the address is a contract
+      assembly {
+        if iszero(extcodesize(tokenAddress)) { revert(0, 0) }
+      }
+      
+      (bool success, bytes memory data) = tokenAddress.call(abi.encodeWithSelector(
```

Comment:
Any reason we don't use encodeWithSignature like the others? It's more readable.

Reply:
I tried, it did not work... I did not push investigations further on this one.

Reply:
EncodeWithSignature should be a short version of what you're doing on the selector.
It would be wise to make sure we understand what we're doing here.

Reply:
Indeed 👍  I got confused by something else apparently. Thanks for the review